File: backend/src/main/java/com/example/backend/domain/vote/dto/VoteRequestDto.java
package com.example.backend.domain.vote.dto;

import com.example.backend.domain.vote.entity.Vote;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class VoteRequestDto {
    private String location;
    private String address;
    private Double latitude;
    private Double longitude;

    public Vote toEntity(Long groupId){
        return Vote.builder()
                .groupId(groupId)
                .location(this.location)
                .address(this.address)
                .latitude(this.latitude)
                .longitude(this.longitude)
//                .voterCount(0)
                .build();
    }
}

File: backend/src/main/java/com/example/backend/domain/vote/dto/VoteRequestDto.kt
package com.example.backend.domain.vote.dto

import com.example.backend.domain.vote.entity.Vote

data class VoteRequestDto(
    val location: String,
    val address: String,
    val latitude: Double,
    val longitude: Double
) {
    fun toEntity(groupId: Long): Vote {
        return Vote(
            groupId = groupId,
            location = this.location,
            address = this.address,
            latitude = this.latitude,
            longitude = this.longitude
        )
    }
}
